src/hotspot/share/jvmci/jvmciEnv.cpp

```diff
@@ -229,17 +235,20 @@ void JVMCIEnv::init_env_mode_runtime(JavaThread* thread, JNIEnv* parent_env) {
 }
 
 JVMCIEnv::JVMCIEnv(JavaThread* thread, JVMCICompileState* compile_state, const char* file, int line):
-    _throw_to_caller(false), _file(file), _line(line), _compile_state(compile_state) {
-  init_env_mode_runtime(thread, NULL);
+    _throw_to_caller(false), _file(file), _line(line), _attach_threw_OOME(false), _compile_state(compile_state) {
+  init_env_mode_runtime(thread, NULL, false);
+  if (_attach_threw_OOME) {
+    compile_state->set_failure(true, "Out of memory");
```

Comment:
Could this be more explicit like `JVMCI compiler attach returned out of memory`?  cc @dougxc

Reply:
I would suggest: `Out of memory while attaching libjvmci to current thread`

Reply:
Fixed.